backend/handlers.go

```diff
@@ -58,35 +56,45 @@ func (cs *server) publishHandler(w http.ResponseWriter, r *http.Request, _ httpr
 		return
 	}
 	defer body.Close()
-
 	// Unmarshal the JSON message into a map
-	var data map[string]interface{}
-	if err := json.Unmarshal(msg, &data); err != nil {
+	decoder := json.NewDecoder(bytes.NewReader(msg))
+	// Create a new map to store the sensorDataMap
+	var sensorIDMap map[string]interface{}
+	// Decode into the sensorDataMap
+	if err := decoder.Decode(&sensorIDMap); err != nil {
 		http.Error(w, "Invalid JSON payload", http.StatusBadRequest)
 		return
 	}
-
-	// Check if "sensorID" key exists
-	if sensorID, ok := data["sensorID"].(string); ok {
-		// Manipulate the "sensorID"
-		sensorID = splitSensorID(sensorID)
-
-
-		// Update the map with the new "sensorID" value
-		data["sensorID"] = sensorID
+	if sensorIDMap == nil {
```

Comment:
Can this be nil here?

Reply:
Nope , no chance . removed this extra check .